core/src/main/java/org/mule/runtime/core/message/ErrorTypeBuilder.java

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.core.message;
+
+import static org.mule.runtime.core.util.Preconditions.checkState;
+
+import org.mule.runtime.api.message.ErrorType;
+
+/**
+ * Builder for {@link ErrorType}. This must be the only mechanism to create an instance
+ * of {@code ErrorType}.
+ *
+ * @since 4.0
+ */
+public final class ErrorTypeBuilder {
+
+  private static final String MULE_NAMESPACE = "mule";
+  private static final String ANY_STRING_REPRESENTATION = "ANY";
+  private static final String GENERAL_STRING_REPRESENTATION = "GENERAL";
+
+  public static final ErrorType ANY =
+      ErrorTypeBuilder.builder().namespace(MULE_NAMESPACE).stringRepresentation(ANY_STRING_REPRESENTATION).build();
+  public static final ErrorType GENERAL = ErrorTypeBuilder.builder().namespace(MULE_NAMESPACE)
+      .stringRepresentation(GENERAL_STRING_REPRESENTATION).parentErrorType(ANY).build();
+
+  private String stringRepresentation;
+  private String namespace;
+  private ErrorType parentErrorType;
+
+  public static ErrorTypeBuilder builder() {
```

Comment:
We can't do that since we need this in a separate class since it is meant to be used internally only and not as part of the API.